synapse/storage/databases/main/stats.py

```diff
@@ -210,34 +210,56 @@ async def update_room_state(self, room_id: str, fields: Dict[str, Any]) -> None:
         * topic
         * avatar
         * canonical_alias
+        * guest_access
 
-        A is_federatable key can also be included with a boolean value.
+        An is_federatable key can also be included with a boolean value.
 
         Args:
             room_id: The room ID to update the state of.
             fields: The fields to update. This can include a partial list of the
                 above fields to only update some room information.
         """
-        # Ensure that the values to update are valid, they should be strings and
-        # not contain any null bytes.
+        # Ensure that the values to update are valid. str values should not
+        # contain any null bytes.
         #
         # Invalid data gets overwritten with null.
         #
         # Note that a missing value should not be overwritten (it keeps the
         # previous value).
-        sentinel = object()
-        for col in (
+        str_columns = [
             "join_rules",
             "history_visibility",
             "encryption",
             "name",
             "topic",
             "avatar",
             "canonical_alias",
-        ):
-            field = fields.get(col, sentinel)
-            if field is not sentinel and (not isinstance(field, str) or "\0" in field):
-                fields[col] = None
+            "guest_access",
+        ]
+        other_columns = [
+            "is_federatable",
+        ]
+
+        state_dict = {}
+        sentinel = object()
+
+        for column_name in fields.keys():
```

Comment:
Seems like it would be simpler to iterate over `str_columns` and then you don't have to check `if column_name in str_columns`.

The current implementation of iterating the keys, then `get` with a sentinel doesn't really make sense.

Reply:
You're right that the sentinel is no longer necessary. Will axe :axe:

Reply:
I think it is still necessary, it is there to differentiate an incoming value in `fields` of `None` from the key not existing in the incoming dict.

Reply:
jk, it is necessary so that we won't override old values.